src/libsyntax/json.rs

```diff
@@ -41,6 +42,7 @@ pub struct JsonEmitter {
     /// Whether "approximate suggestions" are enabled in the config
     approximate_suggestions: bool,
     ui_testing: bool,
+    error_codes: HashSet<String>,
```

Comment:
As I mentioned in passing at the end of [a previous comment](https://github.com/rust-lang/rust/pull/48337#discussion_r171177402), `Handler` [already knows](https://github.com/rust-lang/rust/blob/master/src/librustc_errors/lib.rs#L268-L271) what what codes we've emitted; `JsonEmitter` shouldn't have to know this.

Reply:
Oh indeed. I'll remove this add.